### PR TITLE
Created react app "Moons of the Solar system"

### PR DESCRIPTION
**Pull request by sebalp100**
# Created react app "Moons of the Solar system"

Today we are creating an app called "Moons of the Solar system" for the React-Redux capstone project.

This app implements the following:

- Uses Redux and connects the store with React
- Uses React Router
- A home page that displays the planet's data from the API (https://api.le-systeme-solaire.net/en/)
- A details page that displays the moon's data from the API
- A search planet functionality for the home page using filter()
- A back button functionality for the details page that redirects the user back to the home page
- Followed styles using the template provided
- Test using React testing library
- Follows the single-page application (SPA) approach

Video showcasing the project: https://drive.google.com/file/d/1iT-wuMADv3tQ3TY4LeMHb3-Xwowz45Jw/view?usp=sharing

### CODE REVIEW

File: src/components/Home.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                  value={searchInput}
+                  placeholder="Search Planet"
+                />
+              </div>
+
+              <div className="background">
+                <img className="backImage" alt="icon" src={logo} />
+                <div className="title">
+                  <h1>SOLAR SYSTEM</h1>
+                  <h2>206 moons</h2>
```
Comment (FrederickMih) [inline src/components/Home.js:62]: - [Optional] 206 is manually displayed as compared to what's done on the detail page :smile:

File: src/redux/home/homeSlice.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable no-param-reassign */
```
Comment (FrederickMih) [inline src/redux/home/homeSlice.js:1]: - [Optional] We could have tried to resolve the error flagged by Linters to make sure that our codebase is up to standard :+1: .

File: src/components/Home.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  const dispatch = useDispatch();
+  const [searchInput, setSearchInput] = useState('');
+
+  const handleChange = (e) => {
+    e.preventDefault();
+    setSearchInput(e.target.value);
+  };
+
+  const filter = planets.filter((planet) => planet.name.toLowerCase()
+    .match(searchInput.toLowerCase()));
```
Comment (FrederickMih) [inline src/components/Home.js]: - [ ] Filter works perfectly :+1:  but based on the requirement of this project, this was supposed to be achieved on a component that holds **no state**( **Filter** stateless component).

File: src/components/Home.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+import earth from '../assets/worldwide.png';
+import venus from '../assets/venus.png';
+import mars from '../assets/mars.png';
+import mercury from '../assets/mercury.png';
+import saturn from '../assets/saturn.png';
+import neptune from '../assets/neptune.png';
+import jupiter from '../assets/jupiter.png';
+import uranus from '../assets/uranus.png';
+
+const Home = () => {
```
Comment (FrederickMih) [inline src/components/Home.js:16]: - [Optional] It is always advisable to split the big complex component into a few smaller simpler components so it is easier to understand :+1:

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @sebalp100 , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!




## Highlights :green_circle: 

- PR is professional :+1: 
- Styling of the app followed the design guidelines with a single consistent color :heavy_check_mark:
- Great and meaningful commit messages :heavy_check_mark: 
- Descriptive README.md file :heavy_check_mark: 
- App is deployed properly :heavy_check_mark: 
- All Linter checks passed :heavy_check_mark:  

## Suggested changes

- [x] **Testing:** The app should have unit tests for functions in redux code and integration tests  using React Testing Library(You can perform a mocking data fetching and Redux Store connection)

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by sebalp100** [comment]
Added missing tests

File: src/components/Home.js (hunk 3 above)
Comment (ichala) [inline src/components/Home.js]: - [x] As Mentioned by the previous reviewer, the filtering should be happening in a stateless component, kindly try to fix that 🙏

**Review by ichala — CHANGES_REQUESTED** [review]
Hi Student
Good job so far! 
👏👏👏 
You still need to work on some issues to go to the next project, but you're almost there!

the-simpsons-homer-simpson

## To Highlight 🎉

- Tests Added ✔️ 


 ## Required Changes ♻️

- [x] **_🛑  Check The Comments Below_**





## Optional suggestions

- N/A

**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Home.js (hunk 3 above)
Comment (sebalp100) [reply src/components/Home.js]: Thank you, sorry I forgot to add that component. I will fix it 👍

**Comment by sebalp100** [comment]
A Stateless Filter component has been added 👍

**Review by DammyShittu — APPROVED** [review]
Hi @sebalp100,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sebalp100** [comment]
@DammyShittu @FrederickMih @ichala Thank you for your feedback 👍

**Pull request merged**